BotUI/src/app/api/root/route.ts

```diff
@@ -1,32 +1,29 @@
-import { QueryReq } from "@/app/general/interfaces";
+import { Attribute } from "@/app/general/interfaces";
 import { getOperator } from "@/app/general/utils";
+import { strOrNum } from "@/app/general/types";
 import { NextRequest, NextResponse } from "next/server";
 import fs from "fs";
 import csvParser from "csv-parser";
 
 export async function POST(request: NextRequest) {
     const req = await request.json();
-    const queryReq = req.queryReq as QueryReq[];
+    const attributes = req.queryParams as Attribute[];
     const filePath = req.filePath as string;
-    const rows = await filterCSV(filePath, queryReq);
+    const rows = await filterCSV(filePath, attributes);
     return NextResponse.json(rows);
 }
 
-export async function filterCSV(filePath: string, queryReq: QueryReq[]) {
+async function filterCSV(filePath: string, attributes: Attribute[]) {
     return new Promise<any[]>((resolve, reject) => {
-        const operators = queryReq.map(
+        const operators = attributes.map(
             (query) => getOperator(query.operator) as any
         );
         const rows: any[] = [];
         fs.createReadStream(filePath)
             .pipe(csvParser())
             .on("data", (row: any) => {
-                const isRequired = queryReq.every((query, index) =>
-                    operators[index](
-                        row[query.parameter],
-                        ...query.functionParams
-                    )
-                );
+                const isRequired = isRowRequired(attributes, operators, row);
```

Comment:
nice

Reply:
but why create param and not just add isRowRequired(attributes, operators, row) in an "if" statement

Reply:
done